### PR TITLE
nushellPlugins.dbus: init at 0.10.0

### PR DESCRIPTION
**Pull request by DarkKronicle**
## Description of changes

Add [nu_plugin_dbus](https://github.com/devyn/nu_plugin_dbus)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by tobim — APPROVED** [review]
Hashes are correct, builds fine, and code follows the same patterns as the other plugins.

**Comment by DarkKronicle** [comment]
I'm not sure if this will build on darwin (from what I can tell dbus should work on Mac?), all other plugins had some sort of added dependency. Does anyone know about that? I can wait a few more days for OfBorg to finish.

File: pkgs/shells/nushell/plugins/default.nix [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
+  lib,
+  newScope,
+  IOKit,
+  CoreFoundation,
+  Foundation,
+  Security,
+  dbus,
+}:
+lib.makeScope newScope (
+  self: with self; {
```
Comment (SuperSandro2000) [inline pkgs/shells/nushell/plugins/default.nix:11]:
```suggestion
  self: {
```
this probably avoids the infinite recursion

File: pkgs/shells/nushell/plugins/dbus.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    repo = "nu_plugin_dbus";
+    rev = version;
+    hash = "sha256-Bb55IO/qkQRVkPPyS0iYxYUw6qxhWMuaLj9oxK+I1fk=";
+  };
+
+  passthru.updateScript = gitUpdater { };
+
+  cargoHash = "sha256-XmERSV2ihl2eS3PcT9hrP0PpTKiRic0Yz2UQyN+fqU8=";
+
+  nativeBuildInputs = [ pkg-config ] ++ lib.optionals stdenv.cc.isClang [ rustPlatform.bindgenHook ];
```
Comment (SuperSandro2000) [inline pkgs/shells/nushell/plugins/dbus.nix:28]:
```suggestion
  nativeBuildInputs = [ pkg-config ] ++ lib.optionals stdenv.cc.isClang [ rustPlatform.bindgenHook ];

```

File: pkgs/shells/nushell/plugins/dbus.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  gitUpdater,
+  pkg-config,
+  dbus,
+}:
+let
+  version = "0.9.0";
+in
+rustPlatform.buildRustPackage {
+  inherit version;
+  pname = "nu-plugin-dbus";
```
Comment (SuperSandro2000) [inline pkgs/shells/nushell/plugins/dbus.nix:15]:
```suggestion
}:

rustPlatform.buildRustPackage rec {
  pname = "nu-plugin-dbus";
  version = "0.9.0";
```

File: pkgs/shells/nushell/plugins/default.nix (hunk 1 above)
Comment (DarkKronicle) [reply pkgs/shells/nushell/plugins/default.nix]: That worked, had to add `callPackage` as well.

**Comment by DarkKronicle** [comment]
Rebased and updated to use the newer nushell version. Added IOKit for Darwin because it failed to build, not sure if it needs CoreFoundation or Foundation.

**Comment by DarkKronicle** [comment]
Closed because #349887

**Pull request closed without merging**